lib/gel/environment.rb

```diff
@@ -364,12 +364,22 @@ def self.activate(install: false, output: nil, error: true)
     @active_lockfile = true
     loader = Gel::LockLoader.new(Gel::ResolvedGemSet.load(lockfile), gemfile)
 
+    if lock_outdated?(loaded, loader)
+      write_lock(output: $stderr, lockfile: lockfile)
+    end
+
     base_store = Gel::Environment.store
     base_store = base_store.inner if base_store.is_a?(Gel::LockedStore)
 
     loader.activate(Gel::Environment, base_store, install: install, output: output)
   end
 
+  def self.lock_outdated?(gemfile, lockfile)
+    gemfile_gems = gemfile.gems.map { |gem| gem[0] }
+    lockfile_gems = lockfile.gem_set.dependency_names
+    (lockfile.gem_set.dependency_names & gemfile_gems).size < [gemfile_gems.size, lockfile_gems.size].max
```

Comment:
IIRC at least one of these lists is already sorted -- might be simpler to normalize as needed then compare them directly?

That being said, if we're going to do this (and I agree it's a good idea), I think we need to go a bit further: I believe we want to compare the full `dependencies`, including version constraints therein.

Reply:
You are right. I updated this with @jhawthorn's help to compare what we would generate from the Gemfile to what we have in the dependencies section of the Gemfile.lcok